lidar_odometry:
  ros__parameters:
    # 使用仿真时间
    use_sim_time: true

    # Registration Type
    # 0: Point-to-Point ICP
    # 1: Point-to-Plane ICP (NICP)
    # 2: GICP
    # 3: NDT
    registration_type: 2  # Default: GICP
    
    # Solver Type
    # 0: Ceres
    # 1: GTSAM_SE3
    # 2: GTSAM_SO3_R3
    # 3: GN (Gauss-Newton)
    # 4: PCL
    # 5: Koide
    solve_type: 5  # Default: Koide
    
    # Registration Parameters
    max_correspondence_distance: 1.0  # meters
    max_iterations: 30
    translation_epsilon: 0.001
    rotation_epsilon: 0.001
    
    # NDT Specific Parameters
    resolution: 1.0  # Grid resolution
    
    # Multi-threading Parameters
    num_threads: 4
    
    # Point Cloud Pre-processing
    downsample_leaf_size: 0.4  # meters
    
    # Debug Options
    verbose: false

    # ROS
    lidar_topic: "/points_raw"  # /sensing/lidar/top/rectified/pointcloud
    imu_topic: "/imu_raw"
    ground_truth_path_topic: "/kitti/ground_truth"
    output_odom_topic: "/output_odom"
    output_path_topic: "/output_path"
    output_cloud_topic: "/output_cloud"

    # local map
    local_map_min_frame_size: 5
    update_frame_size: 10
    update_translation_delta: 0.5  # meters
    update_rotation_delta: 15.0  # degrees

    # save map
    save_map_path: "/home/ubuntu/ros_ws/src/optimization_learning/map"

    # calibration
    # KITTI
    # T_imu2lidar: [1.00000, -0.00079, 0.00202, 0.81054,
    #               0.00076, 0.99989, 0.01482, -0.30705,
    #               -0.00204, -0.01482, 0.99989, 0.80272,
    #               0.00000, 0.00000, 0.00000, 1.00000]

    # NCLT
    T_imu2lidar: [1.00000, 0.00000, 0.00000, 0.00000,
                  0.00000, 1.00000, 0.00000, 0.00000,
                  0.00000, 0.00000, 1.00000, 0.28,
                  0.00000, 0.00000, 0.00000, 1.00000]
